hypothesis-python/src/hypothesis/internal/conjecture/junkdrawer.py

```diff
@@ -394,3 +396,21 @@ def find(self, condition: Callable[[T], bool]) -> T:
                 self.__values.append(value)
                 return value
         raise NotFound("No values satisfying condition")
+
+
+_gc_start = 0
+_gc_cumulative_time = 0
+
+
+def _gc_callback(phase, _info):
```

Comment:
This looks like it should be fast, but some of the tests look like they're failing due to timing differences.  Can we confirm that the overall impact is negligible?

Reply:
Could it be because we're running on Schroedinger time in the test suite - time only increases when observed. Previously, gc was unobserved, now time is observed twice, adding 2x `TICK_TIME` to passed time.

If so, this pr effectively adds indeterminism to passed time, and I don't know if that should be handled or if we should just add a hack in `conftest.py` to ignore that time (yet another gc callback)

Reply:
Ouch, yeah, that'd be it.  I _think_ that not-observing gc time under coverage is probably the least-awful option we have... and should be pretty clean if we just add an `and not IN_COVERAGE_TESTS` to the install-hook condition.

(definitely feels like some kind of karma though; we sure have made some choices that we're now living with)

Reply:
How do you feel about this solution: https://github.com/HypothesisWorks/hypothesis/pull/3979/commits/8635813207e3aa7dea65c2ba532803054ffdfc08? I think it will be more maintenable to place the fixup in the same place as the "problem"

Reply:
Doh. `gc.callbacks` doesn't work if reassigned. Fix on the way.